src/visualization/utilities_p4est_t8code.jl

```diff
@@ -0,0 +1,305 @@
+end
+
+function search_points_in_t8code_mesh_3d_callback_element(forest::t8_forest_t,
+                                                          ltreeid::t8_locidx_t,
+                                                          element::Ptr{t8_element_t},
+                                                          is_leaf::Cint,
+                                                          leaf_elements::Ptr{t8_element_array_t},
+                                                          tree_leaf_index::t8_locidx_t)::Cint
+    # Continue the search
+    return 1
+end
+
+function search_points_in_t8code_mesh_3d_callback_query(forest::t8_forest_t,
+                                                        ltreeid::t8_locidx_t,
+                                                        element::Ptr{t8_element_t},
+                                                        is_leaf::Cint,
+                                                        leaf_elements::Ptr{t8_element_array_t},
+                                                        tree_leaf_index::t8_locidx_t,
+                                                        queries_ptr::Ptr{sc_array_t},
+                                                        query_indices_ptr::Ptr{sc_array_t},
+                                                        query_matches::Ptr{Cint},
+                                                        num_active_queries::Csize_t)::Cvoid
+    # It would be nice if we could just use
+    #   t8_forest_element_points_inside(forest, ltreeid, element, coords,
+    #                                   num_active_queries, query_matches, tolerance)
+    # However, this does not work since the coordinates of the points
+    # on the curve are given in physical coordinates but t8code only
+    # knows reference coordinates (in [0, 1]).
+    # Currently, we do not store the `mapping` but only the resulting
+    # `tree_node_coordinates` in the mesh. Thus, we need to use them
+    # to convert the reference coordinates of the `element` to physical
+    # coordinates (assuming polydeg == 1) and then check whether the
+    # coordinates of the points are inside the `element`.
+
+    # Get the references coordinates of the element.
+    # Note: This assumes that we are in 3D and that the element is a hexahedron.
+    user_data = Ptr{Ptr{Float64}}(t8_forest_get_user_data(forest))
+    vertex = PtrArray(unsafe_load(user_data, 2), (3,))
+    eclass_scheme = t8_forest_get_eclass_scheme(forest, T8_ECLASS_HEX)
+    t8_element_vertex_reference_coords(eclass_scheme, element, 0, vertex)
+    r000 = SVector(vertex[1], vertex[2], vertex[3])
+    t8_element_vertex_reference_coords(eclass_scheme, element, 1, vertex)
+    r100 = SVector(vertex[1], vertex[2], vertex[3])
+    t8_element_vertex_reference_coords(eclass_scheme, element, 2, vertex)
+    r010 = SVector(vertex[1], vertex[2], vertex[3])
+    t8_element_vertex_reference_coords(eclass_scheme, element, 4, vertex)
```

Comment:
Honest question: Do we need the vertex with id 4 here and not 3? We have 0, 1, and 2 before. So I'm just wondering.

Reply:
- 0: back, left, bottom
- 1: front, left, bottom
- 2: back, right, bottom
- 3: front, right, bottom
- 4: back, left, top

We need the vertex moving up in z direction. Vertex 3 is still is the xy plane.

Reply:
Ah, I see. That makes sense. Thanks for the explanation!